lib/pagerduty/http_transport.rb

```diff
@@ -30,10 +36,17 @@ def self.http
       http
     end
 
-    def self.transported?(response)
-      response.is_a?(Net::HTTPSuccess) || response.is_a?(Net::HTTPRedirection)
+    def http_proxy
+      Net::HTTP.Proxy(
+        @options[:proxy_host],
+        @options[:proxy_port],
+        @options[:proxy_username],
+        @options[:proxy_password],
+      )
```

Comment:
The `Proxy` method returns a dynamic class that can be used polymorphically as `Net::HTTP`. If the `proxy_host` is `nil` a proxy won't be used.

http://ruby-doc.org/stdlib-1.9.3/libdoc/net/http/rdoc/Net/HTTP.html#method-c-Proxy